ffi/src/lib.rs

```diff
@@ -905,6 +1634,18 @@ pub unsafe extern "C" fn changelist_entry_at(tx_report: *mut TransactionChange,
     tx_report.changes[index].clone()
 }
 
+/// Destructor for releasing the memory of [InProgressBuilder](mentat::InProgressBuilder).
```

Comment:
Why aren't we using `define_destructor!`?  Is it the life times?  You can define multiple rules for a macro to handle additional cases...

Reply:
It was the lifetimes. My macro foo is not very strong but I shall try it out.